spec/interactors/save_person_spec.rb

```diff
@@ -0,0 +1,81 @@
+require 'rails_helper'
+
+RSpec.describe SavePerson do
+  let(:current_location) { create :location, city: 'flint' }
+  let(:current_person)   { create :person, location: current_location }
+
+  let(:params) {{
+    id: current_person&.id,
+    preferred_contact_method: 'email',
+    email: 'we@together.coop',
+    location: {
+      id: current_location&.id,
+      city: '',
+      state: 'DC', # statehood now!
+    },
+  }}
+
+  let(:interaction) { SavePerson.run params }
+  let(:person)      { interaction.result }
+
+  describe 'modification' do
+    it 'updates the current person and associated location' do
+      expect(person).to eq(current_person.reload)
+      expect(person.location).to eq current_location.reload
+    end
+
+    it 'allows for existing fields to be nullified' do
+      expect(person.location.city).to be_empty
+    end
+
+    context 'when params include a person.id but no location.id' do
+      let(:current_location) { nil }
+
+      it 'fails' do
+        expect { interaction }.to raise_error(/Missing location.id/)
```

Comment:
I really like all the test cases you've outlined here. These answered all the questions I had looking at the rest of the code